### PR TITLE
Feature/staking wallet features

### PR DESCRIPTION
**Pull request by OrLavy**
## Description

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. 
If it fixes a `bug` or resolves a `feature request`, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/orbs-network/orbs-contributor-guide/blob/master/CONTRIBUTING.md) doc
- [ ] I have opened an issue and discussed the solution suggested in this PR
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: voting/orbs-pos-data/src/testkit/OrbsTokenServiceMock.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    this.senderAccountAddress = address;
+    return this;
+  }
+
+  // WRITE //
+  approve(spenderAddress: string, amount: number): PromiEvent<TransactionReceipt> {
+    const promitEvent = this.generateTxData();
+    if (this.autoCompleteTxes) {
+      this.completeTx(promitEvent);
+      // TODO : O.L : Check if this is the best place to put this effect
```
Comment (gilamran) [inline voting/orbs-pos-data/src/testkit/OrbsTokenServiceMock.ts]: Implement  your idea regarding  map of effects

File: voting/orbs-pos-data/src/__tests__/orbs-token-service.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  it('should allow overriding of contract address', async () => {
+    const contractAddress = '0xaaaaaE5030f3e769FaC89AEF5ac34EbE8Cf95a76';
+    const localWeb3Mock = new Web3Mock();
+    const localOrbsTokenService = new OrbsTokenService(localWeb3Mock as any, contractAddress);
+
+    expect(localWeb3Mock.eth.Contract).toBeCalledWith(ERC20ContractABI, contractAddress);
+  });
+
+  it('should call "stake" with the amount', async () => {
```
[the remark this answers is not in the document]
Comment (OrLavy) [reply voting/orbs-pos-data/src/__tests__/orbs-token-service.test.ts]: Done

File: voting/orbs-pos-data/src/testkit/StakingServiceMock.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ interface ITxData {
 }
 
 export class StakingServiceMock implements IStakingService {
+  private _stakingContractAddress: string = 'DUMMY_CONTRACT_ADDRESS';
```
[the remark this answers is not in the document]
Comment (OrLavy) [reply voting/orbs-pos-data/src/testkit/StakingServiceMock.ts]: Done

File: voting/orbs-pos-data/src/interfaces/IStakingService.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ export interface IStakingStatus {
 }
 
 export interface IStakingService {
+  stakingContractAddress: string;
```
[the remark this answers is not in the document]
Comment (OrLavy) [reply voting/orbs-pos-data/src/interfaces/IStakingService.ts]: Done

File: voting/orbs-pos-data/src/services/StakingService.ts [hunk 5]
```diff
@@ -15,14 +15,15 @@ import { IStakingService, IStakingStatus } from '../interfaces/IStakingService';
 import { STAKING_CONTRACT_ADDRESS } from '../contracts-adresses';
 
 export class StakingService implements IStakingService {
+  public readonly stakingContractAddress: string;
```
[the remark this answers is not in the document]
Comment (OrLavy) [reply voting/orbs-pos-data/src/services/StakingService.ts]: Done

File: voting/orbs-pos-data/src/testkit/StakingServiceMock.ts [hunk 6]
```diff
@@ -132,6 +133,14 @@ export class StakingServiceMock implements IStakingService {
     this.addressToStakeStatus.set(address, status);
   }
 
+  public setStakingContractAddress(address: string) {
+    this._stakingContractAddress = address;
+  }
+
+  public get stakingContractAddress(): string {
```
[the remark this answers is not in the document]
Comment (OrLavy) [reply voting/orbs-pos-data/src/testkit/StakingServiceMock.ts]: done

File: voting/orbs-pos-data/src/services/OrbsTokenService.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    );
+
+    return async () => {
+      let unsubscribeOfSpecificApprovalPromise = getUnsubscribePromise(specificApprovalEventEmitter);
+
+      return unsubscribeOfSpecificApprovalPromise;
+    };
+  }
+
+  // SUBSCRIPTIONS //
```
[the remark this answers is not in the document]
Comment (OrLavy) [reply voting/orbs-pos-data/src/services/OrbsTokenService.ts]: Done

File: voting/orbs-pos-data/src/testkit/OrbsTokenServiceMock.ts (hunk 1 above)
Comment (OrLavy) [reply voting/orbs-pos-data/src/testkit/OrbsTokenServiceMock.ts]: Done

**Pull request merged**